subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -100,6 +100,41 @@ intended for internal use and was never intended to be used as part of the publi
 
 Starting in Gradle 9.0, this method will be removed without replacement.
 
+[[test_task_default_classpath]]
+==== Relying on Test task default testClassesDirs and classpath
+
+By default, when applying the `JavaPlugin`, standalone `Test` tasks are configured such that their `testClassesDirs`
+and `classpath` mirror the corresponding values in the default `test` `TestSuite`. This behavior will be removed
+in Gradle 9.0.
+
+To restore the existing behavior, use the following code:
+
+====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+tasks.withType<Test>().configureEach {
+    testClassesDirs = tasks.test.get().testClassesDirs
+    classpath = tasks.test.get().classpath
+}
+----
+=====
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+tasks.withType(Test).configureEach {
+    testClassesDirs = tasks.test.testClassesDirs
+    classpath = tasks.test.classpath
+}
+----
+=====
+====
```

Comment:
💭 Maybe add a brief sentence or two about how this old behavior could lead to confusion when creating custom `Test` tasks which don't need these conventions.

Reply:
Thanks for the quick review, though I had accidentally marked this as non-draft. This PR still need a bit of work to fix up some of the samples. If there are any changes worth reviewing I'll let you know once I've updated the samples.